packages/dev/core/src/Shaders/screenSpaceReflection2.fragment.fx

```diff
@@ -0,0 +1,386 @@
+// Screen Space Reflection Post-Process based on the following tutorial:
+// https://lettier.github.io/3d-game-shaders-for-beginners/screen-space-reflection.html
+
+precision highp float;
+uniform sampler2D textureSampler;
+
+#ifdef SSR_SUPPORTED
+uniform sampler2D normalSampler;
+uniform sampler2D positionSampler;
+uniform sampler2D reflectivitySampler;
+uniform sampler2D depthSampler;
+#if defined(BACKUP_TEXTURE)
+    uniform samplerCube backUpSampler;
+#endif // BACKUP
+// SSR parameters
+uniform float maxDistance;
+uniform float resolution;
+uniform int steps;
+uniform float thickness;
+uniform float strength;
+uniform float falloffExponent;
+uniform float roughnessFactor;
+uniform float distanceFade;
+uniform float maxReflectivityForSSRReflections;
+#include<helperFunctions>
+#endif // SSR_SUPPORTED
+
+uniform mat4 view;
+uniform mat4 projection;
+
+// camera properties
+uniform float minZ;
+uniform float maxZ;
+uniform vec3 cameraPos;
+
+// Varyings
+varying vec2 vUV;
+
+#ifdef SSR_SUPPORTED
+
+/**
+ * According to specular, see https://en.wikipedia.org/wiki/Schlick%27s_approximation
+ */
+vec3 fresnelSchlick(float cosTheta, vec3 F0)
+{
+    return F0 + (vec3(1.0) - F0) * pow(1.0 - cosTheta, 5.0);
+}
+
+vec3 hash(vec3 a)
+{
+    a = fract(a * 0.8);
+    a += dot(a, a.yxz + 19.19);   
+    return fract((a.xxy + a.yxx) * a.zyx); 
+}
+
+// Structs
+struct ReflectionInfo {
+    float visibilityBackup;
+    float visibility;
+    vec2 coords;
+    bool miss;
+};
+
+/** Computes and returns the coordinates and the visibility of the reflected pixel if any, as well as a boolean defining if there is a reflected pixel or if it's a miss
+ * The intersection algorithm based on a David Lettier's tutorial uses 2D ray marching 
+ */
+ReflectionInfo getReflectionInfo2DRayMarching(vec3 dirVS, vec3 hitCoordVS, vec2 texSize){
+
+    ReflectionInfo info;
+    // Default values if the algorithm fail to find intersection:
+    info.visibilityBackup = 0.0;
+    info.visibility = 0.0;
+    info.coords = vUV;
+    info.miss = true;
+
+    // Calculate the start and end point of the reflection ray in view space.
+    vec4 startVS = vec4(hitCoordVS, 1.0);
+    vec4 endVS = vec4(hitCoordVS + (dirVS * maxDistance), 1.0);
+
+    #ifdef RIGHT_HANDED_SCENE
+        if (endVS.z > minZ) // no need to compute anything, the max depth of reflection is not in the view space (not behind the near plane)
+    #else 
+        if (endVS.z < minZ) 
+    #endif
+    {
+    #if defined(BACKUP_TEXTURE)
+        info.visibilityBackup = 1.0;
+    #endif
+    return info;
+    }
+
+
+    // Calculate the start and end point of the reflection ray in screen space.
+    vec4 startSS = projection * startVS; // Project to screen space.
+    startSS.xyz /= startSS.w; // Perform the perspective divide.
+    #ifndef WEBGPU

```

Comment:
My mistake, only the Z coordinate is in the 0..1 range in WebGPU, so you can remove the `#ifndef` as x/y also need the remapping in WebGPU.